cuda_core/pyproject.toml

```diff
@@ -46,8 +46,8 @@ dependencies = [
 ]
 
 [project.optional-dependencies]
-cu11 = ["cuda-bindings==11.8.*"]
-cu12 = ["cuda-bindings==12.*"]
+cu11 = ["cuda-bindings[all]==11.8.*"]
+cu12 = ["cuda-bindings[all]==12.*"]
 test = ["cython>=3.0", "setuptools", "pytest>=6.2.4"]
 test-cu11 = ["cuda-core[test]", "cupy-cuda11x", "nvidia-cuda-runtime-cu11"]  # runtime headers needed by CuPy
 test-cu12 = ["cuda-core[test]", "cupy-cuda12x", "nvidia-cuda-runtime-cu12"]  # runtime headers needed by CuPy
```

Comment:
Sorry I have been in a shitty state today and realized this late... 

Following the offline discussion, I believe the root cause of using mismatching NVRTC version is due to the changes in line 49-50, not 52-53. We need to add a conditional branch here:
https://github.com/NVIDIA/cuda-python/blob/0edec40244b4ade5484ca49c3662bd0bf22fa67d/ci/tools/run-tests#L57
- if testing against local CTK, don't install any CTK wheels
- if testing against CTK wheels, make sure we install the right major.minor versions (patch version can be ignored for now)

Let me take care of this change.

Reply:
Seems like it does.